validator/accounts/status.go

```diff
@@ -0,0 +1,240 @@
+	withCert string,
+	grpcHeaders []string,
+	grpcRetries uint) ([]grpc.DialOption, error) {
+	var transportSecurity grpc.DialOption
+	if withCert != "" {
+		creds, err := credentials.NewClientTLSFromFile(withCert, "")
+		if err != nil {
+			return nil, errors.Wrapf(err, "Could not get valid credentials: %v", err)
+		}
+		transportSecurity = grpc.WithTransportCredentials(creds)
+	} else {
+		transportSecurity = grpc.WithInsecure()
+		log.Warn(
+			"You are using an insecure gRPC connection! Please provide a certificate and key to use a secure connection.")
+	}
+
+	if maxCallRecvMsgSize == 0 {
+		maxCallRecvMsgSize = 10 * 5 << 20 // Default 50Mb
+	}
+
+	md := make(metadata.MD)
+	for _, hdr := range grpcHeaders {
+		if hdr != "" {
+			ss := strings.Split(hdr, "=")
+			if len(ss) != 2 {
+				log.Warnf("Incorrect gRPC header flag format. Skipping %v", hdr)
+				continue
+			}
+			md.Set(ss[0], ss[1])
+		}
+	}
+
+	return []grpc.DialOption{
+		transportSecurity,
+		grpc.WithDefaultCallOptions(
+			grpc.MaxCallRecvMsgSize(maxCallRecvMsgSize),
+			grpc_retry.WithMax(grpcRetries),
+			grpc.Header(&md),
+		),
+		grpc.WithStatsHandler(&ocgrpc.ClientHandler{}),
+		grpc.WithBlock(),
+	}, nil
+}
+
+// FetchAccountStatuses fetches validator statuses from the BeaconNodeValidatorClient
+// for each validator public key.
+func FetchAccountStatuses(
+	ctx context.Context,
+	beaconNodeRPCProvider ethpb.BeaconNodeValidatorClient,
+	pubkeys [][]byte) ([]ValidatorStatusMetadata, error) {
+	ctx, span := trace.StartSpan(ctx, "accounts.FetchAccountStatuses")
+	defer span.End()
+	ctx, cancel := context.WithTimeout(ctx, 30*time.Second /* Cancel if running over thirty seconds. */)
+	defer cancel()
+
+	errorChannel := make(chan error, MaxRequestLimit)
+	statusChannel := make(chan []ValidatorStatusMetadata, MaxRequestLimit)
+	// Launch routines to fetch statuses.
+	i, numBatches := 0, 0
+	for ; i+MaxRequestKeys < len(pubkeys); i += MaxRequestKeys {
+		go fetchValidatorStatus(
+			ctx, beaconNodeRPCProvider, pubkeys[i:i+MaxRequestKeys], statusChannel, errorChannel)
+		numBatches++
+	}
+	if i < len(pubkeys) {
+		go fetchValidatorStatus(
+			ctx, beaconNodeRPCProvider, pubkeys[i:], statusChannel, errorChannel)
+		numBatches++
+	}
+	// Wait for fetch routines to finish.
+	var err error
+	allStatuses := make([][]ValidatorStatusMetadata, 0, numBatches)
+	for i := 0; i < numBatches; i++ {
+		select {
+		case statuses := <-statusChannel:
+			allStatuses = append(allStatuses, statuses)
+		case e := <-errorChannel:
+			err = e
+			log.Warnln(err)
+		case <-ctx.Done():
+			return nil, ctx.Err()
+		}
+	}
+	if err != nil {
+		return nil, err
+	}
+	return mergeStatuses(allStatuses), nil
+}
+
+func fetchValidatorStatus(
+	ctx context.Context,
+	rpcProvider ethpb.BeaconNodeValidatorClient,
+	pubkeys [][]byte,
+	statusChannel chan []ValidatorStatusMetadata,
+	errorChannel chan error) {
+	if ctx.Err() == context.Canceled {
+		errorChannel <- errors.Wrap(ctx.Err(), "context has been canceled.")
```

Comment:
No punctuation in error messages
```suggestion
		errorChannel <- errors.Wrap(ctx.Err(), "context has been canceled")
```

Reply:
Code removed.